### PR TITLE
Add Github Actions

### PR DESCRIPTION
**Pull request by ArangoGutierrez**

### CODE REVIEW

**Comment by ArangoGutierrez** [comment]
Can't see why the actions won't trigger....

**Comment by ArangoGutierrez** [comment]
@elezar any clue what am I missing here?

**Review by elezar — DISMISSED** [review]
Lgtm

**Comment by ArangoGutierrez** [comment]
Waiting on https://gitlab.com/nvidia/cloud-native/k8s-kata-manager/-/merge_requests/32

**Comment by ArangoGutierrez** [comment]
@shivamerla any update on moving https://gitlab.com/nvidia/cloud-native/k8s-kata-manager/-/merge_requests/32 to GitHub

File: .github/workflows/golang.yaml [hunk 1]
```diff
@@ -47,7 +45,7 @@ jobs:
         - name: Install Go
           uses: actions/setup-go@v5
           with:
-            go-version: '1.20'
+            go-version-file: 'go.mod'
```
Comment (elezar) [inline .github/workflows/golang.yaml:48]: I would recommend doing something like https://github.com/NVIDIA/k8s-device-plugin/blob/47d1aa131da1ddc017e4b1e2c8133690e1bd94f8/.github/workflows/golang.yaml#L33-L41 here.

We want to ensure that we're using consistent go versions.

File: Makefile [hunk 2]
```diff
@@ -120,36 +120,19 @@ CONTROLLER_GEN = $(PROJECT_DIR)/bin/controller-gen
...
-	@echo "Running 'make $(*)' in docker container $(BUILDIMAGE)"
+$(DOCKER_TARGETS): docker-%:
+	@echo "Running 'make $(*)' in container image $(BUILDIMAGE)"
 	$(DOCKER) run \
 		--rm \
-		-e GOCACHE=/tmp/.cache \
-		-v $(PWD):$(PWD) \
-		-w $(PWD) \
+		-e GOCACHE=/tmp/.cache/go \
+		-e GOMODCACHE=/tmp/.cache/gomod \
```
Comment (elezar) [inline Makefile:128]: I've noticed that we technically need to specify a golangci-lint cache here too.

File: versions.mk [hunk 3]
```diff
@@ -18,4 +18,7 @@ vVERSION := v$(VERSION:v%=%)
 
 CUDA_VERSION := 12.3.1
 
-GOLANG_VERSION := 1.21.5
+GOLANG_VERSION := 1.21.4
```
Comment (elezar) [inline versions.mk:21]: Let's not change the go version as part of this PR.

File: Makefile (hunk 2 above)
Comment (ArangoGutierrez) [reply Makefile:128]: or we could run without docker, as in https://github.com/NVIDIA/k8s-device-plugin/blob/main/.github/workflows/golang.yaml#L42-L47

Comment (elezar) [reply Makefile:128]: Yes, but if I want to run `golangci-lint` after starting an interactive shell with `make .shell`.

Comment (elezar) [reply Makefile:128]: Can be done as a follow-up.

**Review by elezar — APPROVED** [review]

**Pull request merged**